src/riak_cs_gc_manager.erl

```diff
@@ -0,0 +1,285 @@
+%%% gen_fsm callbacks
+%%%===================================================================
+
+init([]) ->
+    process_flag(trap_exit, true),
+    InitialDelay = riak_cs_gc:initial_gc_delay(),
+    State = case riak_cs_gc:gc_interval() of
+                infinity ->
+                    #gc_manager_state{};
+                Interval when is_integer(Interval) ->
+                    Next = riak_cs_gc:timestamp() + InitialDelay,
+                    TimerRef = erlang:send_after(InitialDelay * 1000, self(), {start, []}),
+                    #gc_manager_state{next=Next,
+                                      interval=Interval,
+                                      initial_delay=InitialDelay,
+                                      timer_ref=TimerRef}
+        end,
+    {ok, idle, State};
+init([testing]) ->
+    {ok, idle, #gc_manager_state{}}.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc All gen_fsm:send_event/2 call should be ignored
+idle(_Event, State) ->
+    {next_state, idle, State}.
+running(_Event, State) ->
+    {next_state, running, State}.
+
+%% @private
+%% @doc
+idle({start, Options}, _From, State) ->
+    MaxWorkers =  riak_cs_gc:gc_max_workers(),
+    %% StartKey = proplists:get_value(start, Options,
+    %%                                riak_cs_gc:epoch_start()),
+    BatchStart = riak_cs_gc:timestamp(),
+    %% EndKey = proplists:get_value('end', Options, BatchStart),
+    Leeway = proplists:get_value(leeway, Options,
+                                 riak_cs_gc:leeway_seconds()),
+
+    %% set many items to GCDState here
+    GCDState = #gc_batch_state{
+                  batch_start=BatchStart,
+                  leeway=Leeway,
+                  max_workers=MaxWorkers},
+
+    case riak_cs_gc_batch:start_link(GCDState) of
+        {ok, Pid} ->
+            {reply, ok, running,
+             State#gc_manager_state{gc_batch_pid=Pid,
+                                    current_batch=GCDState}};
+        Error ->
+            {reply, Error, idle, State}
+    end;
+idle(_, _From, State) ->
+    {reply, {error, idle}, idle, State}.
+
+running(cancel, _From, State = #gc_manager_state{gc_batch_pid=Pid}) ->
+    %% stop gc_batch here
+    catch riak_cs_gc_batch:stop(Pid),
+    NextState=schedule_next(State),
+    {reply, ok, idle, NextState#gc_manager_state{gc_batch_pid=undefined}};
+running({finished, Report}, _From, State = #gc_manager_state{batch_history=H}) ->
+    %% Add report to history
+    NextState=schedule_next(State),
+    {reply, ok, idle,
+     NextState#gc_manager_state{gc_batch_pid=undefined,
+                                batch_history=[Report|H]}};
+running(_Event, _From, State) ->
+    Reply = {error, running},
+    {reply, Reply, running, State}.
+
+%% @private
+%% @doc Not used.
+handle_event(_Event, StateName, State) ->
+    {next_state, StateName, State}.
+
+%% @private
+%% @doc
+handle_sync_event({set_interval, Initerval}, _From, StateName, State) ->
```

Comment:
Interesting coincidence. These are the very reason I thought app env was better so that `resume` could resume with the interval value that had been set last.